app/javascript/shared/transfersLayouts.js

```diff
@@ -109,6 +110,30 @@ const buildPlatesMatrix = function(requestsWithPlates, maxPlates, maxWellsPerPla
   return { platesMatrix: platesMatrix, duplicatedRequests: duplicatedRequests }
 }
 
+const buildLibrarySplitPlatesMatrix = function(requestsWithPlates) {
+  const platesMatrix = []
+  const duplicatedRequests = []
+  for (let i = 0; i < requestsWithPlates.length; i++) {
+    const { request, well, plateObj } = requestsWithPlates[i]
+    if (request === undefined) { continue }
+    const wellIndex = nameToIndex(well.position.name, 8)
+    if (platesMatrix[plateObj.index] === undefined) {
+      platesMatrix[plateObj.index] = {}
+    }
+    if (platesMatrix[plateObj.index][request.library_type] === undefined) {
+      platesMatrix[plateObj.index][request.library_type] = []
+    }
+    if (platesMatrix[plateObj.index][request.library_type][wellIndex] === undefined) {
+      platesMatrix[plateObj.index][request.library_type][wellIndex] = requestsWithPlates[i]
+    }
+    else {
+      duplicatedRequests.push(requestsWithPlates[i])
+    }
```

Comment:
Yes, all the code would need some rethinking about how plateMatrix and duplicatedRequests work or if it could be replaced. We can leave it to when we upgrade to vue3 as that will require more changes in code